File: GXpert/GXpert.Web/Modules/ServerTypes/Syllabus/SemesterColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SemesterRow } from "./SemesterRow";

export interface SemesterColumns {
    Id: Column<SemesterRow>;
    CourseTitle: Column<SemesterRow>;
    ClassTitle: Column<SemesterRow>;
    Title: Column<SemesterRow>;
    SortOrder: Column<SemesterRow>;
    Description: Column<SemesterRow>;
}

export class SemesterColumns extends ColumnsBase<SemesterRow> {
    static readonly columnsKey = 'Syllabus.Semester';
    static readonly Fields = fieldsProxy<SemesterColumns>();
}

File: GXpert/GXpert.Web/Modules/Syllabus/Semester/SemesterColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Syllabus.Columns;

[ColumnsScript("Syllabus.Semester")]
[BasedOnRow(typeof(SemesterRow), CheckNames = true)]
public class SemesterColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [QuickFilter]
    public string CourseTitle { get; set; }
    [QuickFilter]
    public string ClassTitle { get; set; }
    [EditLink]
    public string Title { get; set; }
    public short SortOrder { get; set; }
    public string Description { get; set; }


}